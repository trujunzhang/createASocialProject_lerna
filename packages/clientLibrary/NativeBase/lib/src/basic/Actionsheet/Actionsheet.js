// export class ActionSheet {
//   static show: (
//     configuration: {
//       options: string[] | Array<{ text: string; icon?: string; iconColor?: string }>
//       cancelButtonIndex?: number
//       destructiveButtonIndex?: number
//       title?: string
//     },
//     onSelect: (index: number) => void
//   ) => void
// }
//# sourceMappingURL=Actionsheet.js.map