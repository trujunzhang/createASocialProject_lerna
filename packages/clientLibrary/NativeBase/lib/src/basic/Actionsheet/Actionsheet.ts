import * as React from 'react'

export type ActionSheetContainerStyle = Object | number | any[]

export interface ActionSheetContainerProps {
  style?: ActionSheetContainerStyle
  // Others
  autoHide?: boolean
  duration?: any
}

// export class ActionSheet {
//   static show: (
//     configuration: {
//       options: string[] | Array<{ text: string; icon?: string; iconColor?: string }>
//       cancelButtonIndex?: number
//       destructiveButtonIndex?: number
//       title?: string
//     },
//     onSelect: (index: number) => void
//   ) => void
// }
