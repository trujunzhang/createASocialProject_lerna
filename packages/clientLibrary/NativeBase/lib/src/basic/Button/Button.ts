import * as React from 'react'

import { ReactNative, RnViewStyleProp, RnTextStyleProp, BsStyle } from '@appUtils/rnCommonProps'

/**
 * see Widget Button.js
 */
export interface ButtonProps extends ReactNative.TouchableOpacityProps, BsStyle {
  /**
   * Defines button style
   */
  style?: RnViewStyleProp | Array<RnViewStyleProp>
  /**
   * Defines button text style
   */
  textStyle?: RnTextStyleProp
  // onPress?: () => void
  /**
   * Block level button
   */
  block?: boolean
  /**
   * Vertical button
   */
  vertical?: boolean
  /**
   * Button with badges
   */
  badge?: boolean
  //primary?: boolean,
  /**
   * Gives you effect of Icon-buttons.
   * To have button with transparent background, include this prop.
   */
  transparent?: boolean
  //success?: boolean,
  //danger?: boolean,
  // warning?: boolean,
  //info?: boolean,
  color?: string
  /**
   * Applies outline button style.
   */
  bordered?: boolean
  /**
   * Renders button with slightly round shaped edges.
   */
  rounded?: boolean
  /**
   * For large size button
   */
  large?: boolean
  /**
   * For small size button
   */
  small?: boolean
  /**
   * Used for Icon alignment.
   * Aligns icon to the left in button.
   * By default, icons are aligned to the left in button.
   */
  iconLeft?: boolean
  /**
   * Used for Icon alignment.
   * Aligns icon to the right in button.
   */
  iconRight?: boolean
  /**
   * Used to enable/disable icons
   * Icons align in the center of the button.
   */
  icon?: boolean
  /**
   * Disables onPress option for button
   */
  disabled?: boolean
  active?: boolean
  inputButton?: boolean
  full?: boolean
  light?: boolean
  dark?: boolean
  /**
   * [android] colored ripple effect
   */
  androidRippleColor?: string
  /**
   * Segments
   */
  first?: boolean
  last?: boolean
  /**
   * Adds necessary padding when Text button defined in Left / Right of Header (iOS)
   */
  hasText?: boolean
  // Others
  activeOpacity?: any
  onlyCenter?: boolean
  facebook?: boolean
  twitter?: boolean
  svgLeft?: boolean
}
