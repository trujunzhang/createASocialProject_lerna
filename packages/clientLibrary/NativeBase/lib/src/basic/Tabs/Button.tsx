import * as React from 'react'
// const React = require('react')
// const ReactNative = require('react-native')

import { TouchableOpacity, View } from 'react-primitives'
export const Button = (props) => {
  return <TouchableOpacity {...props}>{props.children}</TouchableOpacity>
}

// module.exports = Button;
//export default Button;
