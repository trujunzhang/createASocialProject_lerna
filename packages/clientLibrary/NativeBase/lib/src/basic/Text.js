import * as React from 'react'
import PropTypes from "prop-types";
import { Text as RNText } from "react-primitives";
import { LodashUtils as _ } from '@app/tools'

import { connectStyle } from "@app/native-base-shoutem-theme";
import mapPropsToStyleNames from "../utils/mapPropsToStyleNames";

class Text extends React.Component {
  render() {
    const {
      uppercase,
      children,
    } = this.props;

    let text;
    if (uppercase) {
      text = React.Children.map(children, child => {
        if (_.isString(child)) {
          return _.toUpper(child);
        } else {
          return child;
        }
      })
    } else {
      text = children;
    }

    return (
      <RNText ref={c => (this._root = c)} {...this.props}>
        {text}
      </RNText>
    );
  }
}

Text.propTypes = {
  ...RNText.propTypes,
  uppercase: PropTypes.bool,
  style: PropTypes.oneOfType([PropTypes.object, PropTypes.number, PropTypes.array]),
};

Text.defaultProps = {
  uppercase: false,
};

const StyledText = connectStyle("NativeBase.Text", {}, mapPropsToStyleNames)(Text);

export { StyledText as Text };
