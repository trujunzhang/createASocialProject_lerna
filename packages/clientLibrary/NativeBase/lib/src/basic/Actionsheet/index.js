import * as React from 'react';
import { Modal, Platform, ActionSheetIOS, TouchableOpacity, FlatList, Dimensions } from 'react-primitives';
import { connectStyle } from 'native-base-shoutem-theme';
import { Text } from '@appBasic/Text';
import { Icon } from '@appBasic/Icon';
import { Left } from '@appBasic/Left';
import { Right } from '@appBasic/Right';
import { Body } from '@appBasic/Body';
import { ListItem } from '@appBasic/ListItem';
import { mapPropsToStyleNames } from '@app/tools';
class ActionSheetContainer extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            modalVisible: false,
            items: []
        };
    }
    static show(config, callback) {
        this.actionsheetInstance._root.showActionSheet(config, callback);
    }
    static hide() {
        this.actionsheetInstance._root.hideActionSheet();
    }
    showActionSheet(config, callback) {
        if (Platform.OS === 'ios') {
            if (typeof config.options[0] == 'object') {
                let options = config.options;
                let filtered = options.map((item) => {
                    return item.text;
                });
                config.options = filtered;
                ActionSheetIOS.showActionSheetWithOptions(config, callback);
            }
            else {
                ActionSheetIOS.showActionSheetWithOptions(config, callback);
            }
        }
        else {
            this.setState({
                items: config.options,
                title: config.title,
                message: config.message,
                destructiveButtonIndex: config.destructiveButtonIndex,
                cancelButtonIndex: config.cancelButtonIndex,
                modalVisible: true,
                callback: callback
            });
        }
    }
    hideActionSheet() {
        this.setState({ modalVisible: false });
    }
    componentDidMount() {
        if (!this.props.autoHide && this.props.duration) {
            console.warn(`It's not recommended to set autoHide false with duration`);
        }
    }
    render() {
        return (React.createElement(Modal, { animationType: 'fade', transparent: true, visible: this.state.modalVisible, onRequestClose: () => {
                this.state.callback(this.state.cancelButtonIndex);
                this.setState({ modalVisible: false });
            } },
            React.createElement(TouchableOpacity, { activeOpacity: 1, onPress: () => {
                    this.state.callback(this.state.cancelButtonIndex);
                    this.setState({ modalVisible: false });
                }, style: {
                    backgroundColor: 'rgba(0,0,0,0.4)',
                    flex: 1,
                    justifyContent: 'flex-end'
                } },
                React.createElement(TouchableOpacity, { activeOpacity: 1, style: {
                        backgroundColor: '#fff',
                        minHeight: 56,
                        height: this.state.items.length * 80,
                        maxHeight: Dimensions.get('window').height / 2,
                        padding: 15,
                        elevation: 4
                    } },
                    this.state.title ? React.createElement(Text, { style: { color: '#757575' } }, this.state.title) : null,
                    React.createElement(FlatList, { style: {
                            marginHorizontal: -15,
                            marginTop: this.state.title ? 15 : 0
                        }, data: this.state.items, keyExtractor: (item, index) => String(index), renderItem: ({ index, item }) => {
                            return typeof this.state.items[0] === 'string' ? (React.createElement(ListItem, { onPress: () => {
                                    this.state.callback(parseInt(index));
                                    this.setState({ modalVisible: false });
                                }, style: { borderColor: 'transparent', marginLeft: 14 } },
                                React.createElement(Text, null, item))) : (React.createElement(ListItem, { onPress: () => {
                                    this.state.callback(parseInt(index));
                                    this.setState({ modalVisible: false });
                                }, style: {
                                    borderColor: 'transparent',
                                    marginLeft: 14,
                                    height: 50
                                }, icon: true },
                                React.createElement(Left, null,
                                    React.createElement(Icon, { name: item.icon, style: {
                                            color: item.iconColor ? item.iconColor : undefined
                                        } })),
                                React.createElement(Body, { style: { borderColor: 'transparent', paddingLeft: 7 } },
                                    React.createElement(Text, null, item.text)),
                                React.createElement(Right, null)));
                        } })))));
    }
}
// ActionSheetContainer.propTypes = {
//   ...ViewPropTypes,
//   style: PropTypes.oneOfType([
//     PropTypes.object,
//     PropTypes.number,
//     PropTypes.array
//   ])
// };
const StyledActionSheetContainer = connectStyle('NativeBase.ActionSheetContainer', {}, mapPropsToStyleNames)(ActionSheetContainer);
const ActionSheet = StyledActionSheetContainer;
export { ActionSheet, StyledActionSheetContainer as ActionSheetContainer };
//# sourceMappingURL=index.js.map