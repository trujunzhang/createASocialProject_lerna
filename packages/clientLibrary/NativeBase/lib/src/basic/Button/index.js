import * as React from 'react';
import * as PropTypes from 'prop-types';
import { TouchableOpacity, Platform, View, TouchableNativeFeedback, StyleSheet } from 'react-primitives';
import { connectStyle } from 'native-base-shoutem-theme';
import { platformVariables as variable } from '@appVariables/theme/variables/platform'; // variables // [variable]
import { Text } from '@appBasic/Text';
import computeProps from '@appUtils/computeProps';
import { mapPropsToStyleNames } from '@app/tools';
class Button extends React.Component {
    getInitialStyle() {
        return {
            borderedBtn: {
                borderWidth: this.props.bordered ? 1 : undefined,
                borderRadius: this.props.rounded && this.props.bordered ? variable.borderRadiusLarge : 2
            }
        };
    }
    prepareRootProps() {
        const defaultProps = {
            style: this.getInitialStyle().borderedBtn
        };
        if (Array.isArray(this.props.style)) {
            const flattenedStyle = this.props.style.reduce((accumulator, currentValue) => accumulator.concat(currentValue), []);
            return computeProps({ ...this.props, style: flattenedStyle }, defaultProps);
        }
        return computeProps(this.props, defaultProps);
    }
    render() {
        const variables = this.context.theme
            ? this.context.theme['@@shoutem.theme/themeStyle'].variables
            : variable;
        const children = Platform.OS === 'ios'
            ? this.props.children
            : React.Children.map(this.props.children, (child) => child && child.type === Text
                ? React.cloneElement(child, {
                    uppercase: variables.btnUppercaseAndroidText,
                    ...child.props
                })
                : child);
        if (Platform.OS === 'ios' ||
            Platform.OS === 'web' ||
            variables.androidRipple === false ||
            Platform['Version'] < 21) {
            return (React.createElement(TouchableOpacity, Object.assign({}, this.prepareRootProps(), { ref: (c) => (this._root = c), activeOpacity: this.props.activeOpacity > 0 ? this.props.activeOpacity : 0.5 }), children));
        }
        else {
            if (this.props.rounded) {
                let buttonStyle = { ...this.prepareRootProps().style };
                let buttonFlex = this.props.full || this.props.block ? 1 : buttonStyle.flex;
                return (React.createElement(View, { style: [
                        { maxHeight: buttonStyle.height },
                        buttonStyle,
                        { paddingTop: undefined, paddingBottom: undefined }
                    ] },
                    React.createElement(TouchableNativeFeedback, Object.assign({ ref: (c) => (this._root = c), background: this.props.androidRippleColor
                            ? TouchableNativeFeedback.Ripple(this.props.androidRippleColor, true)
                            : TouchableNativeFeedback.Ripple(variables.androidRippleColor, true) }, this.prepareRootProps()),
                        React.createElement(View, { style: [
                                styles.childContainer,
                                {
                                    paddingTop: buttonStyle.paddingTop,
                                    paddingBottom: buttonStyle.paddingBottom,
                                    height: buttonStyle.height,
                                    flexGrow: buttonFlex
                                }
                            ] }, children))));
            }
            else {
                return (React.createElement(TouchableNativeFeedback, Object.assign({ ref: (c) => (this._root = c), onPress: this.props.onPress, background: this.props.transparent
                        ? TouchableNativeFeedback.Ripple('transparent')
                        : TouchableNativeFeedback.Ripple(variables.androidRippleColor, false) }, this.prepareRootProps()),
                    React.createElement(View, Object.assign({}, this.prepareRootProps()), children)));
            }
        }
    }
}
Button.contextTypes = {
    theme: PropTypes.object
};
// Button.propTypes = {
//   ...TouchableOpacity.propTypes,
//   style: PropTypes.oneOfType([
//     PropTypes.object,
//     PropTypes.number,
//     PropTypes.array
//   ]),
//   block: PropTypes.bool,
//   primary: PropTypes.bool,
//   transparent: PropTypes.bool,
//   success: PropTypes.bool,
//   danger: PropTypes.bool,
//   warning: PropTypes.bool,
//   info: PropTypes.bool,
//   bordered: PropTypes.bool,
//   disabled: PropTypes.bool,
//   rounded: PropTypes.bool,
//   large: PropTypes.bool,
//   small: PropTypes.bool,
//   active: PropTypes.bool
// };
const styles = StyleSheet.create({
    childContainer: {
        flexShrink: 1,
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center'
    }
});
const StyledButton = connectStyle('NativeBase.Button', {}, mapPropsToStyleNames)(Button);
export { StyledButton as Button };
//# sourceMappingURL=index.js.map