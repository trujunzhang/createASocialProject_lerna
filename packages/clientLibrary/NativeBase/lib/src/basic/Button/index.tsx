import * as React from 'react'
import * as PropTypes from 'prop-types'
import {
  TouchableOpacity,
  Platform,
  View,
  TouchableNativeFeedback,
  StyleSheet
} from 'react-primitives'
import { connectStyle } from 'native-base-shoutem-theme'
import { IThemeVariables } from '@app/tools' // typings
import { platformVariables as variable } from '@appVariables/theme/variables/platform' // variables // [variable]

import { Text } from '@appBasic/Text'
import computeProps from '@appUtils/computeProps'

import { mapPropsToStyleNames } from '@app/tools'

import { ButtonProps } from './Button'

class Button extends React.Component<ButtonProps, any> {
  static contextTypes = {
    theme: PropTypes.object
  }

  getInitialStyle() {
    return {
      borderedBtn: {
        borderWidth: this.props.bordered ? 1 : undefined,
        borderRadius: this.props.rounded && this.props.bordered ? variable.borderRadiusLarge : 2
      }
    }
  }

  // _root: React$Element<TouchableOpacity | TouchableNativeFeedback>;
  _root: any

  prepareRootProps() {
    const defaultProps = {
      style: this.getInitialStyle().borderedBtn
    }

    if (Array.isArray(this.props.style)) {
      const flattenedStyle = this.props.style.reduce(
        (accumulator: any, currentValue) => accumulator.concat(currentValue),
        []
      )
      return computeProps({ ...this.props, style: flattenedStyle }, defaultProps)
    }

    return computeProps(this.props, defaultProps)
  }
  render() {
    const variables: IThemeVariables = this.context.theme
      ? this.context.theme['@@shoutem.theme/themeStyle'].variables
      : variable
    const children =
      Platform.OS === 'ios'
        ? this.props.children
        : React.Children.map(this.props.children, (child: any) =>
            child && child.type === Text
              ? React.cloneElement(child, {
                  uppercase: variables.btnUppercaseAndroidText,
                  ...child.props
                })
              : child
          )
    if (
      Platform.OS === 'ios' ||
      Platform.OS === 'web' ||
      variables.androidRipple === false ||
      Platform['Version'] < 21
    ) {
      return (
        <TouchableOpacity
          {...this.prepareRootProps()}
          ref={(c) => (this._root = c)}
          activeOpacity={this.props.activeOpacity > 0 ? this.props.activeOpacity : 0.5}>
          {children}
        </TouchableOpacity>
      )
    } else {
      if (this.props.rounded) {
        let buttonStyle = { ...this.prepareRootProps().style }
        let buttonFlex = this.props.full || this.props.block ? 1 : buttonStyle.flex
        return (
          <View
            style={[
              { maxHeight: buttonStyle.height },
              buttonStyle,
              { paddingTop: undefined, paddingBottom: undefined }
            ]}>
            <TouchableNativeFeedback
              ref={(c) => (this._root = c)}
              background={
                this.props.androidRippleColor
                  ? TouchableNativeFeedback.Ripple(this.props.androidRippleColor, true)
                  : TouchableNativeFeedback.Ripple(variables.androidRippleColor, true)
              }
              {...this.prepareRootProps()}>
              <View
                style={[
                  styles.childContainer,
                  {
                    paddingTop: buttonStyle.paddingTop,
                    paddingBottom: buttonStyle.paddingBottom,
                    height: buttonStyle.height,
                    flexGrow: buttonFlex
                  }
                ]}>
                {children}
              </View>
            </TouchableNativeFeedback>
          </View>
        )
      } else {
        return (
          <TouchableNativeFeedback
            ref={(c) => (this._root = c)}
            onPress={this.props.onPress}
            background={
              this.props.transparent
                ? TouchableNativeFeedback.Ripple('transparent')
                : TouchableNativeFeedback.Ripple(variables.androidRippleColor, false)
            }
            {...this.prepareRootProps()}>
            <View {...this.prepareRootProps()}>{children}</View>
          </TouchableNativeFeedback>
        )
      }
    }
  }
}

// Button.propTypes = {
//   ...TouchableOpacity.propTypes,
//   style: PropTypes.oneOfType([
//     PropTypes.object,
//     PropTypes.number,
//     PropTypes.array
//   ]),
//   block: PropTypes.bool,
//   primary: PropTypes.bool,
//   transparent: PropTypes.bool,
//   success: PropTypes.bool,
//   danger: PropTypes.bool,
//   warning: PropTypes.bool,
//   info: PropTypes.bool,
//   bordered: PropTypes.bool,
//   disabled: PropTypes.bool,
//   rounded: PropTypes.bool,
//   large: PropTypes.bool,
//   small: PropTypes.bool,
//   active: PropTypes.bool
// };

const styles = StyleSheet.create({
  childContainer: {
    flexShrink: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center'
  }
})

const StyledButton = connectStyle('NativeBase.Button', {}, mapPropsToStyleNames)(Button)
export { StyledButton as Button }
