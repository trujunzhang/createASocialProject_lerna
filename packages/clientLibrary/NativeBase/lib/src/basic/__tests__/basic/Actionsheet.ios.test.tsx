import 'react-primitives'
import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { Button } from '../../Button'
import { ActionSheetContainer as ActionSheet } from '../../Actionsheet'
import { Text } from '../../Text'

var BUTTONS = ['Option 0', 'Option 1', 'Option 2', 'Delete', 'Cancel']
var DESTRUCTIVE_INDEX = 3
var CANCEL_INDEX = 4

// Note: test renderer must be required after react-native.

// jest.mock("Platform", () => {
//   const Platform = require.requireActual("Platform");
//   Platform.OS = "ios";
//   return Platform;
// });

it('renders ActionSheet', () => {
  const tree = renderer
    .create(
      <Button
        onPress={() =>
          ActionSheet.show(
            {
              options: BUTTONS,
              cancelButtonIndex: CANCEL_INDEX,
              destructiveButtonIndex: DESTRUCTIVE_INDEX,
              title: 'Select an option'
            },
            (buttonIndex) => {
              this.setState({ clicked: BUTTONS[buttonIndex] })
            }
          )
        }>
        <Text>Actionsheet</Text>
      </Button>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
