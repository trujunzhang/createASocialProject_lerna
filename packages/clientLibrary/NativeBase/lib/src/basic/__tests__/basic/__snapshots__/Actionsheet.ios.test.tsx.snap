// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ActionSheet 1`] = `
<view
  name="View"
  style={
    Object {
      "borderBottomLeftRadius": 2,
      "borderBottomRightRadius": 2,
      "borderTopLeftRadius": 2,
      "borderTopRightRadius": 2,
    }
  }
>
  <view
    name="View"
    style={Object {}}
  >
    <text
      style={Object {}}
    >
      Actionsheet
    </text>
  </view>
</view>
`;
